src/main/java/org/cbioportal/application/rest/vcolumnstore/ColumnStoreStudyController.java

```diff
@@ -85,14 +92,88 @@ public ResponseEntity<List<CancerStudyMetadataDTO>> getAllStudies(
       @Parameter(description = "Name of the property that the result list is sorted by")
           @RequestParam(required = false)
           StudySortBy sortBy,
+      @Parameter(description = "Page size of the result list")
+          @Max(PagingConstants.MAX_PAGE_SIZE)
+          @Min(PagingConstants.MIN_PAGE_SIZE)
+          @RequestParam(required = false)
+          Integer pageSize,
+      @Parameter(description = "Page number of the result list")
+          @Min(PagingConstants.MIN_PAGE_NUMBER)
+          @RequestParam(required = false)
+          Integer pageNumber,
       @Parameter(description = "Direction of the sort") @RequestParam(defaultValue = "ASC")
           Direction direction) {
 
     var sortAndSearchCriteria =
         new SortAndSearchCriteria(
-            keyword, (sortBy != null ? sortBy.getOriginalValue() : ""), direction.toString());
-    return ResponseEntity.ok(
-        CancerStudyMetadataMapper.INSTANCE.toDtos(
-            getCancerStudyMetadataUseCase.execute(projection, sortAndSearchCriteria)));
+            keyword,
+            (sortBy != null ? sortBy.getOriginalValue() : ""),
+            direction.toString(),
+            pageSize,
+            pageNumber);
+
+    var studies = getCancerStudyMetadataUseCase.execute(projection, sortAndSearchCriteria);
+
+    // Pagination should be handled at the DB layer, but currently our query is not
+    // setup to handle this with authorization
+    if (pageSize != null) {
+      studies = studies.stream().limit(pageSize).toList();
+    }
+    var headers = new HttpHeaders();
+    if (projection == ProjectionType.META) {
+      headers.add(HeaderKeyConstants.TOTAL_COUNT, String.valueOf(studies.size()));
+      headers.add(TOTAL_COUNT_HEADER, String.valueOf(studies.size()));
```

Comment:
why do we need two headers for the exact same count? backward compatibility?

Reply:
Yea, one is a standard and the other is legacy. I thought might as well switch to a standard format for displaying counts